Boost: Run the 404 tester when concatenate modules are enabled

donnchawp opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->
Make sure to run the 404 tester each time either of the concatenate modules is enabled. This lets the plugin react more quickly to changes in wp-content/ in case mod_rewrite support has changed.

Fixes #

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Added Activate trait to concatenate classes


### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
pc9hqz-3kc-p2

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->
* Apply the PR
* Add an error_log to the activate functions added to confirm they're called.
* Toggle both concatenate modules off and on again.
* Run tests with `composer phpunit` from projects/plugins/boost/

donnchawp committed:
Run the 404 tester when concatenate modules are enabled

donnchawp committed:
changelog

dilirity reviewed (dismissed):
We had this previously and we removed it. We're just adding it back in. :shipit:

donnchawp committed:
Add a constant to disable 404 tests.

On multisites or sites that can't call themselves this check may not
work correctly, so disable it by setting this constant

donnchawp committed:
Add a comment to trigger checks

donnchawp committed:
Small modification for testing purposes.

donnchawp committed:
Add test for jetpack_boost_404_tester()

donnchawp committed:
Assign '' as PHAN gives out if it's 0 and assigning 0 later.

LiamSarsfield reviewed (approved):
LGTM. Thanks for adding back in the tester upon activation. Unit tests also pass locally for me 👍
Do/should we clear the scheduled cron/event if the user deactivates both the JS and CSS minify?

dilirity reviewed (approved):
Nice work on this! Thanks for adding the tests.

Not sure if you knew this, but you can run tests from an individual file using this command:

```sh
composer phpunit --filter tests/php/lib/minify/test-functions-service.php
```

This will save you time while working on them in the future.

:shipit: 

EDIT:

> Do/should we clear the scheduled cron/event if the user deactivates both the JS and CSS minify?

@LiamSarsfield Yes we should. Good catch!
